<launch>

	<arg name="lvx_file_path" default="livox_test.lvx"/>
	<arg name="bd_list" default="3JEDHB300102021"/> <!-- 3rd floor AVIA: 3JEDHB300102021 4th floor AVIA: 3JEDJ5K00101441--> 
	<arg name="xfer_format" default="0"/>
	<arg name="multi_topic" default="0"/>
	<arg name="data_src" default="0"/>
	<arg name="publish_freq" default="10.0"/>
	<arg name="output_type" default="0"/>
	<arg name="rviz_enable" default="false"/>
	<arg name="rosbag_enable" default="false"/>
	<arg name="cmdline_arg" default="$(arg bd_list)"/>
	<arg name="msg_frame_id" default="livox_frame"/>
	<arg name="lidar_bag" default="true"/>
	<arg name="imu_bag" default="true"/>

	<param name="xfer_format" value="$(arg xfer_format)"/>
	<param name="multi_topic" value="$(arg multi_topic)"/>
	<param name="data_src" value="$(arg data_src)"/>
	<param name="publish_freq" type="double" value="$(arg publish_freq)"/>
	<param name="output_data_type" value="$(arg output_type)"/>
	<param name="cmdline_str" type="string" value="$(arg bd_list)"/>
	<param name="cmdline_file_path" type="string" value="$(arg lvx_file_path)"/>
	<param name="user_config_path" type="string" value="$(find livox_ros_driver)/config/livox_lidar_config.json"/>
	<param name="frame_id" type="string" value="$(arg msg_frame_id)"/>
	<param name="enable_lidar_bag" type="bool" value="$(arg lidar_bag)"/>
	<param name="enable_imu_bag" type="bool" value="$(arg imu_bag)"/>

	<node name="livox_lidar_publisher" pkg="livox_ros_driver"
	      type="livox_ros_driver_node" required="true"
	      output="screen" args="$(arg cmdline_arg)">
		<remap from="/livox/lidar" to="/livox/lidar/old"/>
		<remap from="/livox/imu" to="/livox/imu/old"/>
	</node>

	<group if="$(arg rviz_enable)">
		<node name="rviz" pkg="rviz" type="rviz" respawn="true"
				args="-d $(find livox_ros_driver)/config/display_lidar_points.rviz"/>
    </group>

	<group if="$(arg rosbag_enable)">
    	<node pkg="rosbag" type="record" name="record" output="screen"
          		args="-a"/>
    </group>


	<!-- link lidar to fixed frame (tf does not work properly, so use tf2_ros) -->
	<!-- <node pkg="tf2_ros" type="static_transform_publisher" name="fixed_lidar_frame_broadcaster" args="0.025 0.0 0.05 0 0 0 1 turret/surface_link livox_frame" /> -->

	<!-- fix time stamp issues from livox lidar, see https://github.com/Livox-SDK/livox_ros_driver/issues/43 -->
	<node name="livox_lidar_timefixer" pkg="background_filter" type="livox_lidar_timefixer_node" output="screen" ns="bird_vs_drone" />

    <!-- PointGrey camera lunch: -->
    <!-- Determine this using rosrun pointgrey_camera_driver list_cameras.
    If not specified, defaults to first camera found. -->
    <arg name="camera_name" default="camera" />
    <arg name="camera_serial" default="0" />
    <arg name="calibrated" default="0" />

    <group ns="$(arg camera_name)">
        <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />

        <node pkg="nodelet" type="nodelet" name="camera_nodelet"
            args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
        <param name="frame_id" value="camera" />
        <param name="serial" value="$(arg camera_serial)" />

        <!-- When unspecified, the driver will use the default framerate as given by the
            camera itself. Use this parameter to override that value for cameras capable of
            other framerates. -->
        <!-- <param name="frame_rate" value="15" /> -->

        <!-- Use the camera_calibration package to create this file -->
        <param name="camera_info_url" if="$(arg calibrated)"
                value="file://$(env HOME)/.ros/camera_info/$(arg camera_serial).yaml" />
        </node>

        <node pkg="nodelet" type="nodelet" name="image_proc_debayer"
            args="load image_proc/debayer camera_nodelet_manager">
        </node>
    </group>
</launch>